autodepgraph/node.py

```diff
@@ -86,6 +94,24 @@ def _get_state(self):
             self._state = 'needs calibration'
         return self._state
 
+    def _set_parents(self, val):
+        '''
+        Sets the parents parameter by calling the add_parent method for every
+        item. This ensures that this node is also added to the children of
+        the other nodes.
+        '''
```

Comment:
This function does not set the parameter to equal the value (list of strings) you set unless it was an empty list of parents. 
Instead it adds items to the list of parents. This results in the wrong value being contained in the snapshot (and potentially weird artefacts when saving and plotting the graph). 

I would suggest either making this such that the list after setting corresponds to exactly what is set or removing the set_parents functionality (just like I would remove the set children function completely).